dbt/include/spark/macros/materializations/table.sql

```diff
@@ -41,7 +41,21 @@
 # --- Autogenerated dbt materialization code. --- #
 dbt = dbtObj(spark.table)
 df = model(dbt, spark)
-df.write.mode("overwrite").format("delta").saveAsTable("{{ target_relation }}")
+
+# make sure pandas exists
+import importlib.util
+import pyspark.pandas
+package_name = 'pandas'
+if importlib.util.find_spec(package_name):
+    import pandas
+
+# convert to pyspark.sql.dataframe.DataFrame
+if isinstance(df, pandas.core.frame.DataFrame):
```

Comment:
But this has to happen inside of the `importlib.util.find_spec(package_name):` if right? If not, you will be accessing a module `pandas` that is not really imported if it was not present.